File: primes.py
from math import *

def is_prime(n):
    if n < 4:
        return True
    if n % 2 == 0:
        return False
    for a in range(3, int(sqrt(n))+1):
        if n % a == 0:
            return False
    return True

def calculate_primes(n):
    primes = []
    p = 1
    for i in range(0, n):
        while not is_prime(p):
            p += 1
        primes.append(p)
        p += 1
    return primes

calculate_primes(100000)

File: primes.lua
function is_prime(n)
  if n < 4 then
    return true
  end
  if n % 2 == 0 then
    return false
  end
  for a = 3, math.ceil(math.sqrt(n)) + 1 do
    if n % a == 0 then
      return false
    end
  end
  return true
end

function calculate_primes(n)
  primes = {}
  p = 1
  for i = 0, n do
    while is_prime(p) ~= true do
      p = p + 1
    end
    table.insert(primes, p)
    p = p + 1
  end

  return primes
end

calculate_primes(1000000)
